dbt_invoke/properties.py

```diff
@@ -825,12 +837,33 @@ def _structure_property_file_dict(location, resource_dict, columns_list):
         column_dict = existing_columns_dict.get(
             column, _get_property_column(column)
         )
+        # If dbt_invoke_template.yml exists, adds templated properties to column 
+        # when not already present
+        if template_yml and 'columns' in template_yml.get(resource_type):
+            _apply_template(
+                column_dict, template_yml[resource_type]['columns']
+            )
         property_file_dict[resource_type_plural][0]['columns'].append(
             column_dict
         )
     return property_file_dict
 
 
+def _apply_template(property_dict, template):
+    """
+    Updates a dictionary with template's yml default properties
+    if said dictionary does not already contain the properties.
+
+    :param property_dict: the dictionary to be updated
+    :param template: a dictionary representing the templated properties
+        to be applied on property_dict
+    :return: None
+    """
+    for key in template:
+        if key not in property_dict:
```

Comment:
What if the `key`  is already in the `property_dict`, but we had altered `dbt_invoke_template.yml` between the previous run and the current run?

Reply:
We would keep the value already in the file – else I think it could lead to frustrating behaviors, where dbt-invoke overwrites things. 

In my opinion, The template should only add new lines, and not update any existing ones, if that makes!